schema/validator.go

```diff
@@ -54,6 +55,28 @@ func (v Validator) Validate(src io.Reader) error {
 			"schema %s: unable to validate", v)
 	}
 
+	if v == MediaTypeManifest {
```

Comment:
Alternatively, you could adjust `image-manifest-schema.json` to replace:

    "config": {
      "$ref": "content-descriptor.json"
    },

with a version of descriptor where `mediaType` was an enum listing only `application/vnd.oci.image.config.v1+json`, etc.

Although see [here][1] about the spec not actually defining bounds on the referenced media types.

[1]: https://github.com/opencontainers/image-spec/pull/257/files#r77792057